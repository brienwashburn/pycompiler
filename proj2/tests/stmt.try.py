try: pass
except: pass
except: pass
except: pass
else: pass
finally: pass

try: pass
finally: pass

try:
    try:
        try: pass
        except foo: pass
    except bar as quux:
        bar = 1
    except total:
        total = 1
    finally: blah
except: 
    pass
else:
    break
