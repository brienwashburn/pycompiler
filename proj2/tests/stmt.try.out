(Module
 (Try
  (body (Pass))
  (handlers (except #f #f (Pass)) (except #f #f (Pass)) (except #f #f (Pass)))
  (orelse (Pass))
  (finalbody (Pass)))
 (Try (body (Pass)) (handlers) (orelse) (finalbody (Pass)))
 (Try
  (body
   (Try
    (body
     (Try
      (body (Pass))
      (handlers (except (Name foo) #f (Pass)))
      (orelse)
      (finalbody)))
    (handlers
     (except (Name bar) quux (Assign (targets (Name bar)) (value (Num 1))))
     (except (Name total) #f (Assign (targets (Name total)) (value (Num 1)))))
    (orelse)
    (finalbody (Expr (Name blah)))))
  (handlers (except #f #f (Pass)))
  (orelse (Break))
  (finalbody)))
